src/shims/unix/fd.rs

```diff
@@ -278,6 +278,12 @@ impl WeakFileDescriptionRef {
     }
 }
 
+impl VisitProvenance for WeakFileDescriptionRef {
+    fn visit_provenance(&self, _visit: &mut VisitWith<'_>) {
+        // All our WeakFileDescriptionRef do not have any tags.
```

Comment:
Weak file descriptors are either tracked by a strong ref somewhere, or already dangling and thus not containing anything. So there's no need to visit them here